src/model/config.ts

```diff
@@ -0,0 +1,54 @@
+import BN from "bn.js";
+import { ConfigurableCorePool } from "../core/configurablePool";
+
+export class PoolConfig {
+    id: string;
+    tickSpacing: number;
+    token0: string;
+    token1: string;
+    fee: number;
+}
+
+export class Record {
+    id: string;
+    eventId: number;
+    actionName: string;
+    actionParams: object;
+    timestamp: Date;
+}
+
+export class Snapshot {
+    id: string;
+    nonce: number;
+    baseSnapshot: Snapshot;
+    poolConfig: PoolConfig;
+    records: Record[];
+    baseRecordId: string;
+    lastRecordId: string;
+    used: boolean;
+    timestamp: Date;
+    snapshotStorage: SnapshotStorage;
+
+    static from(id: string): Snapshot {
+        // TODO
+        return new Snapshot();
+    }
+
+    private persist(configurableCorePool: ConfigurableCorePool): string {
+        // TODO
+        return '0';
+    }
+
+}
+
+export class SnapshotStorage {
+    token0Balance: BN;
+    token1Balance: BN;
+    sqrtPriceX96: BN;
+    tickCurrent: number
+    feeGrowthGlobal0X128: BN;
+    feeGrowthGlobal1X128: BN;
+    ticks_json: object;
```

Comment:
If we have to save the binary data of an object into a database, maybe SQL is not the best option.
I am not sure about the upper limit of the number of ticks or positions, but the better practice is definitely using a No-SQL database for directly storing vertically growing objects.

Reply:
will think about it.